waspc/src/Wasp/Generator/WebAppGenerator/Common.hs

```diff
@@ -89,8 +90,15 @@ mkTmplFdWithDstAndData ::
   Path' (Rel WebAppRootDir) File' ->
   Maybe Aeson.Value ->
   FileDraft
-mkTmplFdWithDstAndData srcPathInWebAppTemplatesDir dstPathInWebAppRootDir tmplData =
+mkTmplFdWithDstAndData relSrcPath relDstPath tmplData =
   createTemplateFileDraft
-    (webAppRootDirInProjectRootDir </> dstPathInWebAppRootDir)
-    (webAppTemplatesDirInTemplatesDir </> srcPathInWebAppTemplatesDir)
+    (webAppRootDirInProjectRootDir </> relDstPath)
+    (webAppTemplatesDirInTemplatesDir </> relSrcPath)
     tmplData
+
+mkUniversalTmplFdWithDst :: Path' (Rel UniversalTemplatesDir) File' -> Path' (Rel WebAppRootDir) File' -> FileDraft
```

Comment:
The signature should probably be polymorphic (`File a` and `File b` instead of `File'`), right? The same should apply to the rest of these functions.

@Martinsos am I missing something, I see all old functions use units.